packages/liquidity-widgets/src/components/Widget/index.tsx

```diff
@@ -77,30 +39,21 @@ export default function Widget({
   }, [theme]);
 
   return (
-    <Web3Provider provider={provider || defaultProvider} chainId={chainId}>
+    <WidgetProvider {...props}>
       <TokenListProvider>
-        <WidgetProvider
-          poolAddress={poolAddress}
-          poolType={poolType}
-          positionId={positionId}
-          theme={theme || defaultTheme}
-          feeAddress={feeAddress}
-          feePcm={feePcm}
+        <ZapContextProvider
```

Comment:
@viet-nv Since we decided to implement global state management `Zustand`, we no longer need these 3 context providers. Instead, all the states can be passed through a single store, and all logic you want to add to these 3 providers can be moved to 3 hooks. I think it's a better practice, I will change It in my PR, if you don't have any concerns

Reply:
agree, I keep this one same as previous because I don't want to change too much logic, just wrap all props + pool and position info to zustand, if you have time, you can help me on this